networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1228 @@
+    >>> for network in condensed_networks:
+    ...     print(f'{network}, total weight: {nx.get_node_attributes(network, "weight")}')
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 3, 2: 1}
+    DiGraph with 5 nodes and 4 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1, 4: 1}
+    DiGraph with 4 nodes and 3 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1}
+    DiGraph with 4 nodes and 3 edges, total weight: {0: 1, 1: 1, 2: 1, 3: 1}
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 1, 2: 1}
+    DiGraph with 3 nodes and 2 edges, total weight: {0: 1, 1: 1, 2: 1}
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+    DiGraph with 2 nodes and 1 edges, total weight: {0: 1, 1: 1}
+
+    Notes
+    ------
+    WIP TODO: As multiple independent graphs may form from applying threshold cutoffs to a weighted graph,
+    only the largest is considered. This might be worth considering in re-evaluating the meaning of
+    weighted network hierarchy coordinate evaluations. (See pages 7, 8 of [1]_, supplementary material)
+
+    An threshold_distribution of None results in a linear distribution, otherwise
+     the exponential distribution is sampled from exp(x) \in (0, 1)
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+    """
+    import numpy as np
+
+    if np.array_equal(np.unique(A), [0, 1]):  # binary check
+        num_thresholds = 1
+    else:
+        num_thresholds = num_thresholds
+
+    # Establishing Thresholds
+    if num_thresholds == 1 or np.isclose(np.max(A) - np.min(A), 0, 1e-2):
+        nx_graphs = [nx.from_numpy_matrix(A, create_using=nx.DiGraph)]
+    else:
+        if threshold_distribution is None:
+            try:
+                thresholds = list(
+                    np.round(
+                        np.arange(
+                            np.min(A),
+                            np.max(A),
+                            (np.max(A - np.min(A))) / num_thresholds,
+                        ),
+                        4,
+                    )
+                )  # linear distribution
+            except:
+                thresholds = [np.max(A)] * num_thresholds
+        else:
+            thresholds = _distribute(
+                dist=threshold_distribution,
+                end_value_range=(np.min(A), np.max(A)),
+                n=num_thresholds,
+            )
+        # Converting to binary nx_graphs according to thresholds:
+        nx_graphs = [
+            nx.from_numpy_matrix(np.where(A > threshold, 1, 0), create_using=nx.DiGraph)
+            for threshold in thresholds
+        ]
+    # removes isolated nodes (0 in & out degree) from binary nodes. (not needed for consolidation)
+    for index in range(len(nx_graphs)):
+        nx_graphs[index].remove_nodes_from(list(nx.isolates(nx_graphs[index])))
```

Comment:
```suggestion
    for G in nx_graphs:
        G.remove_nodes_from(nx.isolates(G))
```
You should remove any idioms that use `for index in range(len(...)):`. It can/should almost always be done by iterating over the items themselves. Python makes this faster then the indexes and it is more readable.

Also, can you move the comments from the end of the line to the line previous?  I think that is preferred style for python and it will allow the `black` formatter to put many of these commands on one line too.

Reply:
In-code comments were moved to new lines, which was what I understood to be intended. I fixed the superfluous indices; those that remain are iterating over multiple items. This is not to say that the code could not be optimized considerably; however I have not yet heard from [Professor Solé](http://complex.upf.es/~ricard/Main/RicardSole.html) regarding the implementation of the graph entropy, who I understand to be the intended correspondent for the original paper.